proof.dfy

```diff
@@ -59,10 +59,24 @@ module Proof {
     && (forall commitMsg | && commitMsg in v.network.sentMsgs 
                            && commitMsg.Commit? 
                            && IsHonestReplica(c, commitMsg.sender)
-          :: QuorumOfPreparesInNetwork(c, v, commitMsg.seqID, commitMsg.clientOp) )
+          :: QuorumOfPreparesInNetwork(c, v, commitMsg.view, commitMsg.seqID, commitMsg.clientOp) )
   }
 
-  predicate HonestReplicasLockOnCommitForGivenView(c:Constants, v:Variables) {
+  predicate HonestReplicasLockOnPrepareForGivenView(c: Constants, v:Variables)
```

Comment:
...AgreeOnPrePrepare... instead of lock ?

Reply:
Good point for clarification. There are 2 such predicates:
-  HonestReplicasLockOnPrepareForGivenView - This predicate states that honest replicas accept the first PrePrepare they receive and vote with a Prepare message only for it. The lock on Prepare is meant to highlight the fact that even though a replica can send multiple times a Prepare message for a given Sequence ID for a given View, this message will not change (unless a View Change happens, but this is why we added "ForGivenView" to the predicate name).
-  HonestReplicasLockOnCommitForGivenView - Again if a replica sends a Commit message for a given Sequence ID for a given View it will not change its mind, even if it re-sends this message multiple times it will always be the same for a given View.

We have a separate predicate that states "CommitMsgsFromHonestSendersAgree" - Here we state that all Commit messages generated from honest replicas agree to the assignment of a client operation for a given Sequence ID for a given View.

Reply:
I see, so what's your suggestion?

Reply:
We can add documentation for those predicates to describe what aspect of the honest replicas those predicates capture.

Reply:
Sounds good :) Let's add the comments and merge the PR.